tests/Valkey.Glide.IntegrationTests/ReadFromEndToEndIntegrationTests.cs

```diff
@@ -0,0 +1,992 @@
+        configOptions.EndPoints.Add(TestConfiguration.STANDALONE_HOSTS[0].host, TestConfiguration.STANDALONE_HOSTS[0].port);
+        configOptions.Ssl = TestConfiguration.TLS;
+
+        // Act & Assert: Test invalid assignment through property setter
+        Assert.Throws<ArgumentException>(() =>
+        {
+            configOptions.ReadFrom = new ReadFrom(ReadFromStrategy.Primary, "invalid-az-for-primary");
+        });
+
+        // Test that the configuration remains in a valid state after failed assignment
+        configOptions.ReadFrom = new ReadFrom(ReadFromStrategy.Primary);
+        var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(configOptions);
+        Assert.NotNull(connectionMultiplexer);
+
+        // Test basic functionality
+        var database = connectionMultiplexer.GetDatabase();
+        await database.PingAsync();
+
+        connectionMultiplexer.Dispose();
+    }
+
+    [Fact]
+    public async Task EndToEnd_ConfigurationPipeline_ClonePreservesReadFromConfiguration()
+    {
+        // Arrange
+        var originalConfig = new ConfigurationOptions
+        {
+            ReadFrom = new ReadFrom(ReadFromStrategy.AzAffinity, "us-east-1a")
+        };
+        originalConfig.EndPoints.Add(TestConfiguration.STANDALONE_HOSTS[0].host, TestConfiguration.STANDALONE_HOSTS[0].port);
+        originalConfig.Ssl = TestConfiguration.TLS;
+
+        // Act
+        var clonedConfig = originalConfig.Clone();
+
+        // Modify original to ensure independence
+        originalConfig.ReadFrom = new ReadFrom(ReadFromStrategy.Primary);
+
+        // Connect using cloned configuration
+        var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(clonedConfig);
+
+        // Assert - Verify connection was created successfully
+        Assert.NotNull(connectionMultiplexer);
+
+        // Verify cloned configuration preserved the original ReadFrom settings
+        Assert.NotNull(clonedConfig.ReadFrom);
+        Assert.Equal(ReadFromStrategy.AzAffinity, clonedConfig.ReadFrom.Value.Strategy);
+        Assert.Equal("us-east-1a", clonedConfig.ReadFrom.Value.Az);
+
+        // Verify original configuration was modified independently
+        Assert.NotNull(originalConfig.ReadFrom);
+        Assert.Equal(ReadFromStrategy.Primary, originalConfig.ReadFrom.Value.Strategy);
+
+        // Test basic functionality
+        var database = connectionMultiplexer.GetDatabase();
+        await database.PingAsync();
+
+        // Cleanup
+        connectionMultiplexer.Dispose();
+    }
+
+    #endregion
+
+    #region Performance and Stress Tests
+
+    [Fact]
+    public async Task EndToEnd_PerformanceValidation_MultipleConnectionsWithDifferentReadFromStrategies()
+    {
+        // Test that multiple connections with different ReadFrom strategies can be created efficiently
+        var tasks = new List<Task<ConnectionMultiplexer>>();
+
+        var configurations = new[]
```

Comment:
Make these as test parameters, you already have it in EndToEnd_RoundTripSerialization_ParseToStringToParse_MaintainsConfigurationIntegrity